compiler/Extract.ml

```diff
@@ -1880,99 +1880,102 @@ let extract_global_decl_hol4_opaque (meta : Meta.meta) (ctx : extraction_ctx)
     and {!extract_fun_decl}.
  *)
 let extract_global_decl (ctx : extraction_ctx) (fmt : F.formatter)
-    (global : global_decl) (body : fun_decl) (interface : bool) : unit =
+    (global : global_decl option) (body : fun_decl) (interface : bool) : unit =
```

Comment:
This creates a huge diff for nothing.
Could you have an auxiliary `extract_global_decl_aux` instead (which is the original `extract_global_decl`) and an `extract_global_decl` which has an optional global decl parameter instead?